### ISSUE TITLE
Questions about the test results of iXI dataset

### ISSUE THREAD

**Issue by zhujunkun** [junyuchen245/TransMorph_Transformer_for_Medical_Image_Registration#11]
We are having a bit of a problem running the code  you provided with IXI dataset.

We loaded the 0.744 pre-trained model of the IXI dataset you provided, but the result was only around 0.6.

Just wondering whether  there is any preprocessing required for the IXI dataset。

![result](https://user-images.githubusercontent.com/67640456/160999457-16e972ae-6600-450c-9134-a0be1fd8d1e8.jpg)

Thanks.

**Comment by zhouhanzhouhan** [comment]
Thank the authors for sharing and I want to reproduce train_Transmorph on IXI dataset, but after epoch 100 and the result for the validation set was only 0.41 why is this? On looking at the paper results, it should be at least over 0.6.
Thank you.

**Comment by junyuchen245** [comment]
Hi @zhouhanzhouhan ,

Did you use the dataset we provided [here](https://github.com/junyuchen245/TransMorph_Transformer_for_Medical_Image_Registration/blob/main/IXI/TransMorph_on_IXI.md)? If yes, then this may be due to the PyTorch version. Please check out this issue: https://github.com/junyuchen245/TransMorph_Transformer_for_Medical_Image_Registration/issues/7#issuecomment-1076316337. TransMorph has been tested on Windows PCs running PyTorch 1.7.1+cu110 and 1.11.0+cu113. Additionally, I've been told that PyTorch 1.9 should work.

Thanks,
Junyu

**Comment by tuffr5** [comment]
Hi Junyu,

I have tested the VoxelMorph under the IXI dataset, For which I used Pytorch1.11+cu113 and Pytorch1.9+cu113. It seemed none of them works. When training the model to 100 epoches and the result for the validation set was still around 0.41, and the loss is around -0.059. Same for both cases. Which version of Pytorch are you using to train the VoxelMorph? Can you give some hints of you requirements.txt or env.yml? 

Thank you.

**Comment by junyuchen245** [comment]
Hi @tuffr5 ,

That's interesting. We've never encountered the issue you are facing. Here are the requirements.txt from three of our workstations:
```
future==0.18.2
matplotlib==3.4.3
ml_collections==0.1.0
natsort==7.1.1
numpy==1.21.2
pystrum==0.1
pytorch_msssim==0.2.1
scikit_image==0.18.3
scipy==1.7.1
skimage==0.0
timm==0.4.12
torch==1.11.0+cu113
torchvision==0.12.0+cu113
```
and
```
dominate==2.6.0
future==0.18.2
matplotlib==3.5.1
ml_collections==0.1.0
natsort==7.1.1
numpy==1.21.2
Pillow==9.1.0
pystrum==0.1
requests==2.26.0
scikit_image==0.18.3
scipy==1.7.1
skimage==0.0
timm==0.4.12
torch==1.7.1+cu110
torchvision==0.8.2+cu110
visdom==0.1.8.9
```
and
```
batchgenerators==0.21
dominate==2.6.0
einops==0.3.0
future==0.18.2
hiddenlayer==0.3
matplotlib==3.4.3
ml_collections==0.1.0
natsort==7.1.1
nibabel==3.2.0
nnunet==1.7.0
nnunet.egg==info
numpy==1.18.5
Pillow==9.1.0
pystrum==0.1
requests==2.24.0
scikit_image==0.17.2
scikit_learn==1.0.2
scipy==1.5.2
skimage==0.0
timm==0.4.5
torch==1.7.1+cu110
torchvision==0.8.2+cu110
tqdm==4.51.0
visdom==0.1.8.9
```

All three of them run TransMorph perfectly. Have you tested those pre-trained weights? Are those working?

Thanks,
Junyu

**Comment by tuffr5** [comment]
Thanks for the reply. I have been trying to replicate the result of VoxelMorph (haven't tried Transmorph yet). I have tested pre-trained weights in torch1.9 and it worked. So that is the most confusing part since I only changed the dataset path directory in the train_vxm.py. Will try more maybe.

**Comment by junyuchen245** [comment]
@tuffr5 I'll take a closer look at this on Monday. I'll let you know if I find anything.

**Comment by tuffr5** [comment]
Thanks so much.

**Comment by zhujunkun** [comment]
Sorry, I don't quite understand this part:
I'm using pytorch1.11.0 in Ubuntu system to test this code.
And we are using the pre-trained weights download from your dataset: 
https://github.com/junyuchen245/TransMorph_Transformer_for_Medical_Image_Registration/blob/main/IXI/TransMorph_on_IXI.md#transmorph-variants
![Screenshot from 2022-04-04 11-06-12](https://user-images.githubusercontent.com/67640456/161467430-257be76c-659a-4653-9ed8-4dbe7aa342d0.png)
when running analysis.py we get the result like this.
![Screenshot from 2022-04-04 11-11-09](https://user-images.githubusercontent.com/67640456/161467755-420a9e75-09fe-462d-b769-081a751475ae.png)
I know it should be like this(according to the existing file.
![Screenshot from 2022-04-04 11-14-16](https://user-images.githubusercontent.com/67640456/161468024-1fc76abc-df52-4d54-b437-3323a76ac454.png).
![image](https://user-images.githubusercontent.com/67640456/161468092-2ad230e3-1ebb-4ba3-a156-061cadec6ef6.png)
For the infer_transmorph.py file, we only edit the config here, to make sure that could run.(the default one is "size mismatch")
Thank you for answering.

**Comment by junyuchen245** [comment]
@zhujunkun `analysis.py` was made for test set evaluation, which assumed the input `.csv` files all have 115 entries (i.e., 115 cases in test set):
https://github.com/junyuchen245/TransMorph_Transformer_for_Medical_Image_Registration/blob/f89c4e46ae77e17952da5350a86ac95b1ff0351a/IXI/analysis.py#L30

According to your screenshot, I see that you put validation `.csv` file in the `analysis.py`, which only has 58 entries. `analysis.py` would thus assume the other 115-58=57 entries as all 0s. That's the reason why you are getting that figure.

Junyu

**Comment by zhujunkun** [comment]
@junyuchen245 
Ok. 
I think it works. 
Thanks for your reply.
Junkun.

**Comment by pds001** [comment]
> 感谢作者的分享，我想在 IXI 数据集上重现train_Transmorph，但是在 epoch 100 之后，验证集的结果只有 0.41，为什么会这样？从论文结果来看，它至少应该超过 0.6。 谢谢。

Have you solved it yet? I had the same problem。Sincerely hope for your reply。
